experimental/examples/token/Token.idr

```diff
@@ -36,30 +43,91 @@ getAccountBalance address accounts = case lookup address accounts of
                       Nothing => 0
                       (Just account) => balance account
 
+total getAccountAllowance : (address : Address) -> SortedMap Address Account -> SortedMap Address Nat
+getAccountAllowance address accounts = case lookup address accounts of
+                      Nothing => empty
+                      (Just account) => allowance account
+
 ||| performTransfer transfers tokens from the from address to the dest address.
 ||| @from the address the tokens to be transferred from
 ||| @dest the address the tokens to be transferred to
 ||| @tokens the amount of tokens to be transferred
+||| @storage the current storage
 total performTransfer : (from : Address) -> (dest : Address) -> (tokens : Nat) -> (storage : Storage) -> Either Error Storage
 performTransfer from dest tokens storage =
   let fromBalance = getAccountBalance from (accounts storage)
       destBalance = getAccountBalance dest (accounts storage) in
         case lte tokens fromBalance of
              False => Left NotEnoughBalance
              True =>
-               let accountsStored = insert from (MkAccount (minus fromBalance tokens) (allowance (getAccount from (accounts storage)))) (accounts storage) in
+               let accountsStored = insert from (MkAccount (minus fromBalance tokens) (getAccountAllowance from (accounts storage))) (accounts storage) in
```

Comment:
Is there a way we can look up the from & to account records and update them instead of looking it up multiple times?

Reply:
Yes! I've added `modifyBalance`.